serve/mlc_serve/model/model_common.py

```diff
@@ -0,0 +1,249 @@
+from typing import List, Union, Optional
+
+import structlog
+import numpy as np
+import torch
+import tvm
+
+from .paged_cache_manager import CacheManager
+from ..engine import (
+    SamplingType,
+    SamplingParams,
+)
+
+LOG = structlog.stdlib.get_logger(__name__)
+
+
+def get_gpu_memory(gpu: int = 0) -> int:
+    return torch.cuda.get_device_properties(gpu).total_memory
```

Comment:
we have here cuda usage several times, it means that model is not generic, but depends on nvidia gpu. let's rename model_common to cuda/nvidia dependent name

Reply:
We depend on CUDA for other reasons as well, such as attention and KV cache kernels. Moreover, there is not a good motivation for supporting other environments. So CUDA is practically a requirement and there is nothing wrong about it.

Reply:
Attention and kv_cahce kernels is a part of the model. The paged cache manager is abstract.